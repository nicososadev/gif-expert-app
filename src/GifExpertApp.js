import React, { Fragment, useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Coding']);

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ul>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ul>
        </Fragment>
    )
}

export default GifExpertApp;