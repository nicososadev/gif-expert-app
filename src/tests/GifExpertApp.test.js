import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas en el la app GifExpertApp', () => {
    
    test('Debe mostrarse correctamente la App', () => {
        
        const wrapper = shallow( <GifExpertApp /> )

        expect( wrapper ).toMatchSnapshot()
    })
    
})
