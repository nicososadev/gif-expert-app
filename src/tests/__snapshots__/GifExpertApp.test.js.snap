// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el la app GifExpertApp Debe mostrarse correctamente la App 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Coding"
      key="Coding"
    />
  </ul>
</Fragment>
`;
